### PR TITLE
Refactored logging for consistency

### PR DESCRIPTION
**Pull request by dbwiddis**
### Description

Adds consistent treatment for log error messages across the repo.  
 - Standardizes the log message to match the value returned to the user (without exception details) 
 - Adds exception stack traces in unexpected error situations
 - Adds more details (steps, ids, etc) when possible.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by amitgalitz — DISMISSED** [review]
Let’s review the reason for all the stack trace logging removal again. Would be very hard to debug code without this

**Comment by dbwiddis** [comment]
Flipping to draft mode for further discussion.  I like the standardization that this PR does, and we can decide when we want to add `, exception` to `log.error()` or similar and apply it consistently.

**Comment by dbwiddis** [comment]
@amitgalitz can you please re-review?

> Merging is blocked
> Merging can be performed automatically once the requested changes are addressed.

File: src/main/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandler.java [hunk 1]
```diff
@@ -180,19 +180,20 @@ public void initFlowFrameworkIndexIfAbsent(FlowFrameworkIndex index, ActionListe
...
+                        logger.info("create index: {}", indexName);
                         internalListener.onResponse(true);
                     } else {
                         internalListener.onResponse(false);
                     }
                 }, e -> {
-                    logger.error("Failed to create index {}", indexName, e);
-                    internalListener.onFailure(new FlowFrameworkException(e.getMessage(), ExceptionsHelper.status(e)));
+                    String errorMessage = "Failed to create index " + indexName;
+                    logger.error(errorMessage);
```
Comment (amitgalitz) [inline src/main/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandler.java:190]: do we not want to log the stack trace in cases like this so we can get more information on why the create index failed?

Comment (dbwiddis) [reply src/main/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandler.java:190]: Good catch. I did a global search-and-replace of `logger.error(errorMessage)` in the transport actions but missed those outside it.

File: src/main/java/org/opensearch/flowframework/workflow/DeleteConnectorStep.java [hunk 2]
```diff
@@ -69,8 +69,9 @@ public void onResponse(DeleteResponse deleteResponse) {
 
             @Override
             public void onFailure(Exception e) {
-                logger.error("Failed to delete connector");
-                deleteConnectorFuture.onFailure(new FlowFrameworkException(e.getMessage(), ExceptionsHelper.status(e)));
+                String errorMessage = "Failed to delete connector";
```
Comment (amitgalitz) [inline src/main/java/org/opensearch/flowframework/workflow/DeleteConnectorStep.java:72]: should we add connector ID here? similar for other deletion steps like agent ID and etc.

File: src/main/java/org/opensearch/flowframework/workflow/UndeployModelStep.java [hunk 3]
```diff
@@ -82,8 +82,9 @@ public void onResponse(MLUndeployModelsResponse mlUndeployModelsResponse) {
 
             @Override
             public void onFailure(Exception e) {
-                logger.error("Failed to unldeploy model");
-                undeployModelFuture.onFailure(new FlowFrameworkException(e.getMessage(), ExceptionsHelper.status(e)));
+                String errorMessage = "Failed to undeploy model";
```
[the remark this answers is not in the document]
Comment (amitgalitz) [reply src/main/java/org/opensearch/flowframework/workflow/UndeployModelStep.java:85]: or should we relay on ml-commons stack trace if we there is some exception from there client?

**Review by amitgalitz — APPROVED** [review]
Thanks for making all these changes across the repo!

**Pull request merged**